SofaGLFW/src/SofaGLFW/SofaGLFWBaseGUI.h

```diff
@@ -135,11 +132,21 @@ class SOFAGLFW_API SofaGLFWBaseGUI
     int m_lastWindowPositionY{ 0 };
     int m_lastWindowWidth{ 0 };
     int m_lastWindowHeight{ 0 };
-
-    bool m_isMouseInteractionEnabled { false };
-    
-    std::shared_ptr<sofaglfw::BaseGUIEngine> m_guiEngine;
-
+    double lastProjectionMatrix[16];
+    double lastModelviewMatrix[16];
+    bool m_isMouseInteractionEnabled{ false };
+    float viewPortPositionX {0};
+    float viewPortPositionY {0};
+    float winPositionX {0};
+    float winPositionY {0};
+    SofaGLFWMouseManager* m_sofaGLFWMouseManager;
```

Comment:
I suggest this to be an object, not a pointer.

```suggestion
    SofaGLFWMouseManager m_sofaGLFWMouseManager;
```